note

```diff
@@ -32,62 +41,67 @@ Arguments:
   -e | --edit <DATE: year-month-day>  Open a specific note for editing.
   -p | --print                        Print the contents of a note.
   -c | --create                       Create a note but don't open it for editing.
-  -i | --info                         Print information about a note.\n"
+  -n | --name                         Set filename for note. Will be created in \$NOTE_DIR
+                                      Don't forget an extension like .md
+  -i | --info                         Print information about a note.
+
+The script loads configuration variables from \$XDG_CONFIG_HOME/note/noterc.
+
+Example:
+# Directory where the current note should be stored
+NOTE_DIR=\"\$HOME/notes/\$YEAR/\$MONTH\"
+
+# Name of the Note
+NOTE_NAME=\"\$DAY.md\"\n"
 }
 
-open_note () {
+open_note() {
     if [[ $EDITOR = *"vim"* ]] || [[ $EDITOR = *"nvim"* ]]; then
         # Open Vim or Neovim in insert mode.
-        $EDITOR "+normal G$" +startinsert! $note_path
+        $EDITOR "+normal G$" +startinsert! "$NOTE_DIR/$NOTE_NAME"
     elif [[ $EDITOR = *"emacs"* ]]; then
         # Open Emacs with cursor at EOF.
-        emacs -nw $note_path --eval "(goto-char (point-max))"
+        emacs -nw "$NOTE_DIR/$NOTE_NAME" --eval "(goto-char (point-max))"
     elif [[ $EDITOR = "" ]]; then
         # If no default editor, use Vim and open in insert mode.
-        vim "+normal G$" +startinsert! $note_path
+        vim "+normal G$" +startinsert! "$NOTE_DIR/$NOTE_NAME"
     else
-        $EDITOR $note_path
+        $EDITOR "$NOTE_DIR/$NOTE_NAME"
     fi
 }
 
-year=`date +'%Y'`
-month=`date +'%m'`
-day=`date +'%d'`
-note_path=~/notes/$year/$month/$day.md
-
-if (( $# > 0 )); then
-    while [[ $# -gt 0 ]]
-    do
+if (($# > 0)); then
+    while [[ $# -gt 0 ]]; do
         key="$1"
         case $key in
-            -e|--edit)
+        -e | --edit)
             printf "(e)dit is not yet implemented :(\n"
-            #shift
-            #shift
             exit 0
             ;;
-            -p|--print)
-            cat $note_path
-            shift
+        -p | --print)
+            cat "$NOTE_DIR/$NOTE_NAME"
             ;;
-            -c|--create)
+        -c | --create)
             create_note
-            shift
```

Comment:
Getting rid of the `shift` while handling input means it can't handle more than one argument (For example I couldn't do `note -c -n mynote.md` or `note -n mynote.md -c` to create and not open a new note called `mynote.md`).

I'm not totally opposed to removing the `shift` from some of these arguments as sometimes it doesn't make sense to handle more than one argument (eg -h, -p, -i). But need to think about it a bit.

Reply:
You're right about the shift. Will think about it. Maybe use -n only to give a name for the other options. But then it does not open the document. It needs to check if -n is the only option and if it's the only option, only then to also open the newly created document.